src/main/java/org/terasology/launcher/util/DownloadUtils.java

```diff
@@ -204,8 +204,11 @@ public static boolean isJenkinsAvailable() {
                     throw new ConnectException();
                 }
             }
-        } catch (Exception e) {
+        } catch (IOException e) {
+            e.printStackTrace();
             logger.warn("Could not connect to Jenkins at {}", JENKINS_URL);
+        } catch (Exception e) {
```

Comment:
I had a closer look at what exceptions can occur for what reason:

- `MalformedURLException` in case our static string is not a valid URL - should never happen, and is caught by `Exception` anyways
- `IOException` can occur for multiple reasons (`.openConnection()`, `.getResponseCode()`, …)
- `Exception` due to `AutoClosable` interface - we have to catch this

Reply:
I agree with you. Note that `MalformedURLException` is a child of `IOException`, so the first block will handle it actually. The `catch(Exception)` block is present exclusively due to the `AutoCloseable` object, and it'll be triggered only when `HttpURLConnection::disconnect` goes wrong (which itself throws `IOException`).